### DISCUSSION TITLE
efficient way to send multiple commands

### DISCUSSION THREAD

**Discussion by fabbermen** [StephanJoubert/home_assistant_solarman#636] [Q&A]
I need to change a few registries every hour, and I would like to know the most efficient way to do so.
With efficient I mean that no command will timeout, the sequence takes a limited time (1 minute), and the code is easy to implement in HA.
Do you know if there's any queueing mechanism running in the background that is buffering the command requests?
Or the commands can be sent in one shot with a special syntax?
If you are able to understand how it works, would you suggest what's the best way to send these 3 commands, for example:
1) send one command at the time, with a delay in between
`  - service: solarman.write_multiple_holding_registers
    data:
      register: 244
      values: 2
  - delay:
      hours: 0
      minutes: 0
      seconds: 16
      milliseconds: 0
  - service: solarman.write_multiple_holding_registers
    data:
      register: 268
      values: 9
  - delay:
      hours: 0
      minutes: 0
      seconds: 16
      milliseconds: 0
  - service: solarman.write_multiple_holding_registers
    data:
      register: 243
      values: 1`

2) shoot the commands without any delay, it will be fine
3) special syntax to modify the 3 registries with just one command.

Thanks a million for your help!

**Comment by davidrapan** [comment]
Hi @fabbermen, you can try [ha-solarman](https://github.com/davidrapan/ha-solarman) and:
1. (and 2.) Fire as many commands as you like in sequence, without delay and they will all go through 100%.
3. You can change multiple registers in continuous row. So for example you can change 243 and 244 in single command.
